relay-server/src/service.rs

```diff
@@ -220,7 +222,7 @@ impl ServiceState {
         );
         ProjectCacheService::new(
             config.clone(),
-            buffer_guard.clone(),
+            MemoryChecker::new(memory_stat.clone(), config.clone()),
```

Comment:
Instead of creating new memory checkers, we could just clone a single one and pass that around and internally create the `MemoryStat` in the constructor.

Reply:
We could also do that yes, we will likely not use `MemoryStat` in isolation, so it's fair to implement it this way.